base/src/Data/Macaw/AbsDomain/JumpBounds.hs

```diff
@@ -661,6 +661,45 @@ addRangePred cns v p =
         _ ->
           Right $ branchAssignRangePred n p
 
+-- | @assertEqPred cns x w c isTrue@ asserts equality @x = BVValue c w@ is either true, or false,
+-- based on `isTrue`.
+assertEqPred ::
+  ( OrdF (ArchReg arch)
+  , HasRepr (ArchReg arch) TypeRepr
+  , MemWidth (ArchAddrWidth arch)
+  , ShowF (ArchReg arch)
+  ) =>
+  -- | Constraints that let us evaluate `x`
+  BlockIntraStackConstraints arch ids ->
+  -- | Value to be constrained
+  Value arch ids (BVType w) ->
+  -- | Bitsize of bitvector c
+  NatRepr w ->
+  -- | Numeric constant x should be equal/different to
+  Integer ->
+  -- | `True` if we should assert equality, `False` if we should assert non-equality
+  Bool ->
+  Either String (BranchConstraints arch ids)
+assertEqPred cns x w c isTrue
+  | isTrue =
+    addRangePred cns (intraStackValueExpr cns x)
+                     (mkRangeBound w (fromInteger c) (fromInteger c))
```

Comment:
For symmetry with the `False` case, leave a comment here saying how `x == c` translates to a range bound. I _think_  it's something like `c <= x && x <= c`, but please double-check my understanding.